wpilibjIntegrationTests/src/main/java/edu/wpi/first/wpilibj/EncoderTest.java

```diff
@@ -41,98 +32,125 @@ protected Logger getClassLogger() {
    * Collection of Arrays. For each Array in the Collection, each array element corresponds to a
    * parameter in the constructor.
    */
-  @Parameters
   public static Collection<Integer[]> generateData() {
     return TestBench.getEncoderDIOCrossConnectCollection();
   }
 
-  /**
-   * Constructs a parametrized Encoder Test.
-   *
-   * @param inputA The port number for inputA
-   * @param outputA The port number for outputA
-   * @param inputB The port number for inputB
-   * @param outputB The port number for outputB
-   * @param flip whether or not these set of values require the encoder to be reversed (0 or 1)
-   */
-  public EncoderTest(int inputA, int outputA, int inputB, int outputB, int flip) {
-    m_inputA = inputA;
-    m_inputB = inputB;
-    m_outputA = outputA;
-    m_outputB = outputB;
-
-    // If the encoder from a previous test is allocated then we must free its
-    // members
-    if (encoder != null) {
-      encoder.teardown();
-    }
-    m_flip = flip == 0;
-    encoder = new FakeEncoderFixture(inputA, outputA, inputB, outputB);
-  }
-
-  @AfterClass
-  public static void tearDownAfterClass() {
+  @AfterAll
+  public static void tearDownAfterAll() {
     encoder.teardown();
     encoder = null;
   }
 
   /** Sets up the test and verifies that the test was reset to the default state. */
-  @Before
+  @BeforeEach
   public void setUp() {
     encoder.setup();
-    testDefaultState();
   }
 
-  @After
+  @AfterEach
   public void tearDown() {
     encoder.reset();
   }
 
-  /** Tests to see if Encoders initialize to zero. */
-  @Test
-  public void testDefaultState() {
+  /**
+   * Tests to see if Encoders initialize to zero.
+   *
+   * @param inputA The port number for inputA
+   * @param outputA The port number for outputA
+   * @param inputB The port number for inputB
+   * @param outputB The port number for outputB
+   * @param flip whether or not these set of values require the encoder to be reversed (0 or 1)
+   */
+  @ParameterizedTest
+  @MethodSource("generateData")
+  public void testDefaultState(int inputA, int outputA, int inputB, int outputB, boolean flip) {
+    init(inputA, outputA, inputB, outputB, flip);
     int value = encoder.getEncoder().get();
-    assertTrue(errorMessage(0, value), value == 0);
+    assertTrue(value == 0, errorMessage(0, value, inputA, outputA, inputB, outputB));
```

Comment:
Why is this `assertTrue()` and not `assertEqual()`? I think I've seen this in other files as well, the same goes for them. 
Together with the `name` attribute, perhaps this can make the `errorMessage()` function redundant?

Reply:
I was focusing on porting the tests to JUnit 5 as is before doing cleanups.